client/lib/trigram/test/index.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Internal dependencies
+ */
+import { cosineSimilarity, trigrams, gramsToLookup, lookupToMagnitude } from '../';
```

Comment:
I know everyone has a different opinion on this, so feel free to just ignore this, but my 0.02 is that it's better only unit-test the functions which are intended for the rest of the system to use. You're exporting cosineSimilarity, so the tests should interact with cosineSimilarity. The tests should interact with your code using the same interface that the rest of the system uses.

The idea is that the tests should validate behavior and not implementation. I would say these things are behavior:
- Equal Strings should have cosineSimilarity of ~1
- Completely different strings should have cosineSimilarity of ~0
- Somewhat overlapping strings should have a cosineSimilarity between 0 and 1

I would say this is implementation:
- Generate Trigrams: Hello
```
const expected = [ '_BEGIN_He', 'Hel', 'ell', 'llo', 'lo_END_' ];
```

^^ We could still have an accurate cosine similarity using a different implementation. For example, maybe I'd prefer to go from trigrams to quadrigams(?). Or maybe I want to change the storage mechanism we're using for the trigrams. If you put a test on trigrams, I would be prevented from doing that. I would argue that once we have to refactor a test, they no longer help verify the implementation is accurate.

[This video](https://www.youtube.com/watch?v=EZ05e7EMOLM) goes into a lot more depth about these types of unit testing practices.

But again, feel free to merge this and disregard my comment, I basically just wanted a good reason to post the video :)